File: java/webflux-security-example/src/main/java/com/van1164/webfluxsecurityexample/security/ServerAuthenticationSuccessHandlerImpl.java
package com.van1164.webfluxsecurityexample.security;

import com.van1164.webfluxsecurityexample.user_info.JwtTokenProvider;
import com.van1164.webfluxsecurityexample.user_info.PrincipalDetails;
import com.van1164.webfluxsecurityexample.user_info.TokenInfo;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.server.DefaultServerRedirectStrategy;
import org.springframework.security.web.server.WebFilterExchange;
import org.springframework.security.web.server.authentication.ServerAuthenticationSuccessHandler;
import org.springframework.stereotype.Component;
import org.springframework.web.server.ServerWebExchange;
import org.springframework.web.util.UriComponentsBuilder;
import reactor.core.publisher.Mono;

import java.net.URI;

@Component(value = "authenticationSuccessHandler")
@Slf4j
@AllArgsConstructor
public class ServerAuthenticationSuccessHandlerImpl implements ServerAuthenticationSuccessHandler {
    private JwtTokenProvider jwtTokenProvider;

    @Override
    public Mono<Void> onAuthenticationSuccess(WebFilterExchange webFilterExchange, Authentication authentication) {
        PrincipalDetails principal = (PrincipalDetails) authentication.getPrincipal();
        String userName = principal.getUsername();
        TokenInfo jwt = jwtTokenProvider.createToken(userName);
        return Mono.just(jwt)
                .doOnNext(token -> log.info("Enter SuccessHandler"))
                .flatMap(token -> {
                    // Set the address to be redirected if the login is successful.
                    //로그인 성공시 리다이렉트할 주소 설정.
                    URI uri = UriComponentsBuilder.newInstance().path("/").queryParam("token", token.getToken()).build().toUri();
                    ServerWebExchange exchange = webFilterExchange.getExchange();
                    return new DefaultServerRedirectStrategy().sendRedirect(exchange, uri);
                });
    }

}

File: kotlin/webflux-security-example/src/main/kotlin/com/van1164/webfluxsecurityexample/security/ServerAuthenticationSuccessHandlerImpl.kt
package com.van1164.webfluxsecurityexample.security

import com.van1164.webfluxsecurityexample.logger
import com.van1164.webfluxsecurityexample.user_info.JwtTokenProvider
import com.van1164.webfluxsecurityexample.user_info.PrincipalDetailsReactive
import org.springframework.security.core.Authentication
import org.springframework.security.web.server.DefaultServerRedirectStrategy
import org.springframework.security.web.server.WebFilterExchange
import org.springframework.security.web.server.authentication.ServerAuthenticationSuccessHandler
import org.springframework.stereotype.Component
import org.springframework.web.util.UriComponentsBuilder
import reactor.core.publisher.Mono


@Component(value = "authenticationSuccessHandler")
class ServerAuthenticationSuccessHandlerImpl(
    val jwtTokenProvider: JwtTokenProvider,
) : ServerAuthenticationSuccessHandler {
    override fun onAuthenticationSuccess(
        webFilterExchange: WebFilterExchange,
        authentication: Authentication
    ): Mono<Void> {
        val principal = authentication.principal as PrincipalDetailsReactive
        val userName = principal.username
        val jwt = jwtTokenProvider.createToken(userName)
       return Mono.just(jwt)
           .doOnNext {
               logger.info{"Enter SuccessHandler"}
           }
            .flatMap {
                // Set the address to be redirected if the login is successful.
                //로그인 성공시 리다이렉트할 주소 설정.
                val uri = UriComponentsBuilder.newInstance().path("/").queryParam("token",jwt.token).build().toUri()
                val exchange = webFilterExchange.exchange
                DefaultServerRedirectStrategy().sendRedirect(exchange,uri)
            }
    }
}